client/components/AutoCompleteAgentWithoutExtension.tsx

```diff
@@ -0,0 +1,71 @@
+import { PaginatedSelectFiltered } from '@rocket.chat/fuselage';
+import { useDebouncedValue } from '@rocket.chat/fuselage-hooks';
+import React, { FC, memo, useMemo, useState } from 'react';
+
+import { ILivechatAgent } from '../../definition/ILivechatAgent';
+import { useRecordList } from '../hooks/lists/useRecordList';
+import { AsyncStatePhase } from '../lib/asyncState';
+import { useAvailableAgentsList } from './Omnichannel/hooks/useAvailableAgentsList';
+
+type AutoCompleteAgentProps = {
+	onChange: (value: string) => void;
+	empty: boolean;
+	haveAll?: boolean;
+	value?: string;
+	currentExtension?: string;
+};
+
+const AutoCompleteAgentWithoutExtension: FC<AutoCompleteAgentProps> = (props) => {
+	const { value, currentExtension, onChange = (): void => undefined, haveAll = false } = props;
+	const [agentsFilter, setAgentsFilter] = useState('');
+
+	const debouncedAgentsFilter = useDebouncedValue(agentsFilter, 500);
+
+	const { itemsList: AgentsList, loadMoreItems: loadMoreAgents } = useAvailableAgentsList(
+		useMemo(
+			() => ({ text: debouncedAgentsFilter, includeExtension: currentExtension, haveAll }),
+			[currentExtension, debouncedAgentsFilter, haveAll],
+		),
+	);
+	const { phase: agentsPhase, items: agentsItems, itemCount: agentsTotal } = useRecordList(AgentsList);
+	const sortedByName = agentsItems
+		.sort((a, b) => {
+			if (value === 'all') {
+				return -1;
+			}
+
+			if (a?.username?.localeCompare(b?.username || '')) {
+				return 1;
+			}
+			if (b?.username?.localeCompare(b?.username || '')) {
+				return -1;
+			}
+
+			return 0;
+		})
+		.map((agent): ILivechatAgent & { label: string; value: string } => ({
+			...agent,
+			label: agent?.username || '',
+			value: agent?.username || '',
+		}));
+
+	return (
+		<PaginatedSelectFiltered
+			value={value}
+			onChange={onChange}
+			flexShrink={0}
+			filter={agentsFilter}
+			// @ts-expect-error
```

Comment:
@MartinSchoeler , I'm not really sure how to get rid of this error lol can you help me?

Reply:
@tiagoevanp , your help here is also appreciated 👀